armi/physics/neutronics/crossSectionSettings.py

```diff
@@ -53,19 +55,55 @@
 CONF_XS_EXECUTE_EXCLUSIVE = "xsExecuteExclusive"
 CONF_XS_PRIORITY = "xsPriority"
 
-# These may be used as arguments to ``latticePhysicsInterface._getGeomDependentWriters``.
-# This could be an ENUM later.
+
+class XSGeometryTypes(Enum):
+    """
+    Data structure for storing the available geometry options
+    within the framework.
+    """
+
+    ZERO_DIMENSIONAL = 1
+    ONE_DIMENSIONAL_SLAB = 2
+    ONE_DIMENSIONAL_CYLINDER = 4
+    TWO_DIMENSIONAL_HEX = 8
+
+    @classmethod
+    def _mapping(cls):
+        mapping = {
+            cls.ZERO_DIMENSIONAL: "0D",
+            cls.ONE_DIMENSIONAL_SLAB: "1D slab",
+            cls.ONE_DIMENSIONAL_CYLINDER: "1D cylinder",
+            cls.TWO_DIMENSIONAL_HEX: "2D hex",
+        }
+        return mapping
+
+    @classmethod
+    def getStr(cls, typeSpec: Enum):
+        """
+        Return a string representation of the given ``typeSpec``.
+
+        Examples
+        --------
+        XSGeometryTypes.getStr(XSGeometryTypes.ZERO_DIMENSIONAL) == "0D"
+        XSGeometryTypes.getStr(XSGeometryTypes.TWO_DIMENSIONAL_HEX) == "2D hex"
+        """
+        geometryTypes = [typ for typ in list(cls)]
+        if typeSpec not in geometryTypes:
+            raise TypeError(f"{typeSpec} not in {geometryTypes}")
+        return cls._mapping()[cls[typeSpec.name]]
+
+
 XS_GEOM_TYPES = {
-    "0D",
-    "1D slab",
-    "1D cylinder",
-    "2D hex",
+    XSGeometryTypes.getStr(XSGeometryTypes.ZERO_DIMENSIONAL),
+    XSGeometryTypes.getStr(XSGeometryTypes.ONE_DIMENSIONAL_SLAB),
+    XSGeometryTypes.getStr(XSGeometryTypes.ONE_DIMENSIONAL_CYLINDER),
+    XSGeometryTypes.getStr(XSGeometryTypes.TWO_DIMENSIONAL_HEX),
 }
 
 # This dictionary defines the valid set of inputs based on
 # the geometry type within the ``XSModelingOptions``
 _VALID_INPUTS_BY_GEOMETRY_TYPE = {
-    "0D": {
+    XSGeometryTypes.getStr(XSGeometryTypes.ZERO_DIMENSIONAL): {
```

Comment:
So clearly this works, but I'm curious. Why is this more desirable than the string approach?

Reply:
The string is defined in one place and then if we ever change it in the future it would automatically be updated here. I think it's the same as the argument for using `CONF_XXX` instead of putting the string setting name directly in the code.

Reply:
Since it's used in multiple places this makes more sense now. Thanks!